test/sequential/test-child-process-emfile.js

```diff
@@ -9,6 +9,11 @@ if (common.isWindows) {
   return;
 }
 
+if (process.env.LSAN_OPTIONS) {
+  console.log('1..0 # Skipped: RLIMIT_NOFILE prevents lsan from working');
+  return;
+}
```

Comment:
Maybe you can stuff the file descriptors in an array and `fs.closeSync()` them on exit?